src/agent/src/config.rs

```diff
@@ -236,6 +263,17 @@ impl FromStr for AgentConfig {
         );
         config_override!(agent_config_builder, agent_config, guest_components_procs);
         config_override!(agent_config_builder, agent_config, image_registry_auth);
+        config_override!(
```

Comment:
These `config_override!` should be conditionally compiled with `#[cfg(feature = "guest-pull")]` too, right?

Reply:
yes! done